bin/eslint-dependency-graph.js

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env node
+
+const {CLIEngine} = require('eslint')
+
+// TODO: Figure out how to deactive other rules.
+let cli = new CLIEngine({
+  rules: {
+    'github/dependency-graph': 1
+  }
+})
+cli.executeOnFiles(process.argv.slice(2))
```

Comment:
Step 1 runs eslint on the codebase with the dependency graph rule enabled. The dependency graph rule scans for import/exports and records them as a side effect of running the linter.

Reply:
Where are results recorded if a new `CLIEngine` instance is created each time?

Reply:
The side effect state is recorded in a global `dependencyGraph` map.

https://github.com/github/eslint-plugin-github/blob/b88619b0de7e5ea16173769c3954783359b734b4/lib/rules/dependency-graph.js#L5

The second phase uses this helper method to access the state.

https://github.com/github/eslint-plugin-github/blob/b88619b0de7e5ea16173769c3954783359b734b4/lib/rules/dependency-graph.js#L89